### PR TITLE
feature: stacked graphs

### PR DESCRIPTION
**Pull request by crcro**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

as per: #7409 

before:
screen shot 2017-11-14 at 3 43 01 am

after: 
screen shot 2017-11-14 at 3 43 19 am

### CODE REVIEW

**Comment by murrant** [comment]
You should be using Config instead of a custom function.  Like `Config::get('graphs.port.stacked')`.
You should allow this to be set in the webui, under the webui -> graph settings section.

Also, you probably shouldn't call it MRTG graphs, just stacked graphs is fine.

File: config.php.default [hunk 1]
```diff
@@ -49,3 +49,6 @@ $config['enable_billing'] = 1;
 
 # Enable the in-built services support (Nagios plugins)
 $config['show_services'] = 1;
+
+# Enable the MRTG style graphs if set to true
+$config['graph']['stacked'] = false;
```
Comment (laf) [inline config.php.default]: Maybe `$config['graphs']['stacked'] = false; `

We already have graphs for a few other config options.

Comment (laf) [reply config.php.default]: In fact, I'd say remove it from here anyway as we want users to use the webui to config this

File: html/pages/settings/webui.inc.php [hunk 2]
```diff
@@ -24,6 +24,10 @@
             'svg' => 'svg',
         ),
     ),
+    array('name' => 'webui.graph_stacked',
```
Comment (laf) [inline html/pages/settings/webui.inc.php]: You've used two variables now for this. I'd actually say `graphs.stacked` is better but I realise I used `webui.graph_type` for svg so I'd be fine with keeping this as well.

File: html/includes/graphs/generic_multi_seperated.inc.php [hunk 3]
```diff
@@ -1,19 +1,41 @@
...
+ * @copyright  2017 LibreNMS
+ * @author     LibreNMS Contributors
+*/
+
+use LibreNMS\Config;
 
 require 'includes/graphs/common.inc.php';
 
+if (Config::get('graph.stacked') == true) {
+    $transparency = 95;
```
Comment (laf) [inline html/includes/graphs/generic_multi_seperated.inc.php]: What's the reason for the different transparency values in the files?

Overall we should put this block into a function to make it re-usable

Comment (crcro) [reply html/includes/graphs/generic_multi_seperated.inc.php]: this graph is used on device ports overview and lower values make it unpleasant to view (too much transparency).

Comment (laf) [reply html/includes/graphs/generic_multi_seperated.inc.php]: Is that not the same for other pages then?

**Comment by laf** [comment]
image


image

**Comment by crcro** [comment]
is better this way?
about the colors ... should change them?

**Comment by laf** [comment]
Better but I would still use a key value return on the array, `return array('transparency' => x, 'conversion' => y);` or something so that using the variables within the graph code is obvious to someone else without having to look at what the function does.

The colours definitely need fixing, it's indistinguishable right now on what is what.

**Comment by crcro** [comment]
screen shot 2017-11-16 at 1 17 32 pm
screen shot 2017-11-16 at 1 17 50 pm


is this acceptable @laf ?

**Comment by laf** [comment]
It's definitely better but why can't we just have the original colours with some transparency?

**Comment by crcro** [comment]
because if they overlay you will get that greyed tone (like in your picture where you have same in/out):

screen shot 2017-11-16 at 2 01 38 pm
screen shot 2017-11-16 at 2 01 49 pm

**Comment by laf** [comment]
> because if they overlay you will get that greyed tone (like in your picture where you have same in/out):

Ah sorry, right so you've not changed the colours it's just how it looks because of the transparency.

I'm not the target audience for this really so I'll leave it up to others comment on if they think it looks ok..

**Comment by kkrumm1** [comment]
are there other color options for stacked?

**Comment by crcro** [comment]
@kkrumm1 ... the scope is to use the same colors for inverted and stacked graphs. currently libre uses greens and blues for bits graph but at @laf points i've come with greens and purples ...

**Comment by kkrumm1** [comment]
Looks good to me then :) thank you @crcro

**Comment by kkrumm1** [comment]
👍

**Comment by kkrumm1** [comment]
When I try enable stacked graphs in the WebUI I get this error.
image

**Comment by laf** [comment]
If this is on a test install you try different PRs on then it might be because your schema was already at or past this version. Run this in mysql:

```
INSERT INTO `config` (`config_name`, `config_value`, `config_default`, `config_descr`, `config_group`, `config_group_order`, `config_sub_group`, `config_sub_group_order`, `config_hidden`, `config_disabled`) VALUES('webui.graph_stacked', 'false', 'false', 'Display stacked graphs instead of inverted graphs', 'webui', 0, 'graph', 0, '0', '0');
```

**Comment by kkrumm1** [comment]
That worked... Also, this was on my production server. I never applied this PR to it this was after daily updates.

**Pull request merged**